tests/data/tokenizers/bert_tokenizer_utils_test.py

```diff
@@ -62,13 +63,24 @@ def test_wordpiece_tokenizer(self):
         tokenizer = WordpieceTokenizer(vocab=vocab, unk_token="[UNK]")
 
         self.assertListEqual(tokenizer.tokenize(""), [])
+        self.assertListEqual(tokenizer.tokenize("", with_span=True), [])
 
         self.assertListEqual(
             tokenizer.tokenize("unwanted running"),
             ["un", "##want", "##ed", "runn", "##ing"])
+        self.assertListEqual(
+            tokenizer.tokenize("unwanted running", with_span=True), [
+                ('un\t0\t2'), ('##want\t2\t6'),
```

Comment:
Such string structure looks a little weird to me. But if this is the expected structure what @hunterhector or @ZhitingHu wanted for the project, I am OK to merge this PR.

Reply:
It looks like this implementation tries to get around the typing issue by converting integers to string. I agree with pengzhi that this looks weird. The behavior of this function would be quite hard to understand for the users.

The typing issues should be directly fixable without doing this.
1. This change [here](https://github.com/asyml/texar-pytorch/pull/332/commits/4c290a890f5c9697485bf231af05f33ec8ef3ac2#diff-e5889098da537d273c5122fc8b281b9738038774b79baeada05c7e4141c73760) forgot to import Union
2. You should be able to make the return type of `TokenizerBase` more general, the implemented function can return a more specific type if they wanted to. (https://mypy.readthedocs.io/en/latest/common_issues.html#incompatible-overrides). I.e. you can return `Union[List[str], Tuple[List[str, int, int]]]` in the base, and the implementer class can return either the same, or simply `List[str]`.  
3. You cannot extend the return type from `List[Union[str, Tuple[str, int, int]]]` to `List[Tuple[str, int, int]]`, nor can you use `List[Tuple[str, int, int]]` as a return type.  Because, in Python, mutable List is invariant, which means when you extend it, you cannot change the inner type of it.  The element type inside the list must be the same.  Now you are changing the inner type from `Union[str, Tuple[str, int, int]]` to `Tuple[str, int, int]`, making it more specific. This is only allowed for `covariant` types. I think here you can change the parent type to `Union[List[str], List[Tuple[str, int, int]]`, and then extend it with `Union[List[Tuple[str, int, int]]`, because `Union` is clearly covariant (i.e. you can change the types inside Union to be more specific) in return types.

Even with very complex typing cases, it is probably still better to use `type: ignore` sometimes than using string to represent numbewrs.